ocean_provider/validation/RBAC.py

```diff
@@ -77,6 +77,40 @@ def get_additional_dids(self):
             for additional_input in additional_inputs
         ]
 
+    def get_data(self):
+        if "data" not in self.request.keys():
+            raise Exception("Data to encrypt is empty.")
+        if (
+            not self._check_if_asset()
+            and not self._check_if_respects_file_encryption_schema()
+        ):
+            raise Exception("Invalid type of data.")
+
+        return self.request["data"]
+
+    def _check_if_asset(self):
+        data = self.request["data"]
```

Comment:
I would add a preliminary check: if it is not dict, nor string, return False directly, to avoid if imbrication later